extensions-contrib/azure-extensions/src/test/java/org/apache/druid/storage/azure/AzureCloudBlobIteratorTest.java

```diff
@@ -0,0 +1,291 @@
+
+    EasyMock.expect(resultSegmentPrefixWithCloudBlobsAndDirectories.getContinuationToken())
+            .andReturn(nullResultContinuationToken);
+    EasyMock.expect(resultSegmentPrefixWithCloudBlobsAndDirectories.getResults())
+            .andReturn(resultBlobItemsPrefixWithCloudBlobsAndDirectories);
+
+    EasyMock.expect(storage.listBlobsWithPrefixInContainerSegmented(
+        CONTAINER1,
+        PREFIX_ONLY_CLOUD_BLOBS,
+        nullResultContinuationToken,
+        MAX_LISTING_LENGTH
+    )).andReturn(resultSegmentPrefixOnlyCloudBlobs1);
+
+
+    EasyMock.expect(storage.listBlobsWithPrefixInContainerSegmented(
+        CONTAINER1,
+        PREFIX_ONLY_CLOUD_BLOBS,
+        resultContinuationPrefixOnlyCloudBlobs,
+        MAX_LISTING_LENGTH
+    )).andReturn(resultSegmentPrefixOnlyCloudBlobs2);
+
+    EasyMock.expect(storage.listBlobsWithPrefixInContainerSegmented(
+        CONTAINER1,
+        PREFIX_WITH_NO_BLOBS,
+        nullResultContinuationToken,
+        MAX_LISTING_LENGTH
+    )).andReturn(resultSegmentPrefixWithNoBlobs);
+
+    EasyMock.expect(storage.listBlobsWithPrefixInContainerSegmented(
+        CONTAINER1,
+        PREFIX_WITH_CLOUD_BLOBS_AND_DIRECTORIES,
+        nullResultContinuationToken,
+        MAX_LISTING_LENGTH
+    )).andReturn(resultSegmentPrefixWithCloudBlobsAndDirectories);
+
+    replayAll();
+
+    azureCloudBlobIterator = new AzureCloudBlobIterator(
+        storage,
+        blobItemDruidFactory,
+        PREFIXES,
+        MAX_LISTING_LENGTH
+    );
+
+    List<CloudBlobHolder> expectedBlobItems = ImmutableList.of(
+        cloudBlobDruidPrefixWithOnlyCloudBlobs1,
+        cloudBlobDruidPrefixWithOnlyCloudBlobs2,
+        cloudBlobDruidPrefixWithCloudBlobsAndDirectories
+    );
+    List<CloudBlobHolder> actualBlobItems = new ArrayList<>();
+    while (azureCloudBlobIterator.hasNext()) {
+      actualBlobItems.add(azureCloudBlobIterator.next());
+    }
+    Assert.assertEquals(expectedBlobItems.size(), actualBlobItems.size());
+    Assert.assertTrue(expectedBlobItems.containsAll(actualBlobItems));
+    verifyAll();
+  }
+
+  @Test(expected = NoSuchElementException.class)
+  public void test_next_emptyPrefixes_throwsNoSuchElementException()
+  {
+    azureCloudBlobIterator = new AzureCloudBlobIterator(
+        storage,
+        blobItemDruidFactory,
+        EMPTY_URI_PREFIXES,
+        MAX_LISTING_LENGTH
+    );
+    azureCloudBlobIterator.next();
+  }
+
+  @Test(expected = RE.class)
+  public void test_fetchNextBatch_exceptionThrownInStorage_throwsREException() throws Exception
+  {
+    EasyMock.expect(storage.listBlobsWithPrefixInContainerSegmented(
+        EasyMock.anyString(),
+        EasyMock.anyString(),
+        EasyMock.anyObject(),
+        EasyMock.anyInt()
+    )).andThrow(new URISyntaxException("", ""));
+    azureCloudBlobIterator = new AzureCloudBlobIterator(
+        storage,
+        blobItemDruidFactory,
+        PREFIXES,
+        MAX_LISTING_LENGTH
+    );
+  }
+
```

Comment:
Is there a test that verifies we've read all the blobs from a prefix if there is a continuation token?
Is there a test for a prefix that has nothing in it? A prefix that has only directories?